### PR TITLE
Disable XML in conntrack parsing

### PR DESCRIPTION
**Pull request by 2opremio**
This attempts to improve the performance problems from #1991 

Parsing the default output of conntrack instead of XML turns out to be more difficult than we thought since some fields are optional and context-based.

I put together this hacky non-optimal parser to see if it's better than XML. If there's a considerable difference I will improve it and write a proper parser, otherwise I may just bite the bullet and fix https://github.com/vishvananda/netlink/issues/171

### CODE REVIEW

**Comment by 2opremio** [comment]
After running this in WeaveCloud dev I can see a parsing improvement of 1-(1.74/1.01)*100 ~= 40%

This is a rough estimation and I should make further tests but it's promising.

Before (look for `xml.(*Decoder).unmarshal` taking 1.74s out of 30s)

before


After (look for `decodeStreamedFlow` taking 1.01s out of 30s)

after

**Comment by 2opremio** [comment]
I have made some more thorough testing: multiple runs, using a more loaded machine (similar to the one seen on https://github.com/weaveworks/scope/pull/1985#issuecomment-258826860) and making sure I always used probe pods running in the same machine. The results are even better than I thought:  1-(1.25/4.23) = 70% improvement.


Before:

before

After (focusing on `decodeStreamedFlow`):

after_streamed_2


So, I will clean this up, fix the tests and merge it.

File: probe/endpoint/conntrack.go [hunk 1]
```diff
@@ -204,39 +201,113 @@ func (c *conntrackWalker) run() {
...
-		return
-	}
-
-	defer log.Infof("contrack exiting")
+	scanner := bufio.NewScanner(bufio.NewReader(stdout))
+	defer log.Infof("conntrack exiting")
 
-	// Now loop on the output stream
-	decoder := xml.NewDecoder(reader)
+	// Lop on the output stream
```
Comment (ekimekim) [inline probe/endpoint/conntrack.go]: nit: typo `Lop`->`Loop`

**Comment by 2opremio** [comment]
> Have you double-checked this output against various verisons of the conntrack CLI to ensure it will work with older versions?

We control the version (i.e. the CLI included in the Scope container).

> Is this format guarenteed to not change in the future?

It hasn't changed in a long time. The last conntrack-tools release is from 2012 http://conntrack-tools.netfilter.org/

**Comment by 2opremio** [comment]
> It hasn't changed in a long time. The last conntrack-tools release is from 2012 http://conntrack-tools.netfilter.org/

That's a lie, http://www.netfilter.org/projects/conntrack-tools/downloads.html says otherwise. So, no guarantees but I would doubt it (it isn't very heavily changed https://git.netfilter.org/conntrack-tools/log/src/conntrack.c )

**Comment by alban** [comment]
> > Have you double-checked this output against various verisons of the conntrack CLI to ensure it will work with older versions?
>
> We control the version (i.e. the CLI included in the Scope container).

@2opremio unfortunately, the output depends not only on the version of conntrack CLI but also on the kernel and on the sysctl options configured (see #2117)

**Pull request merged**